### PR TITLE
ci: create pr and release checklist comment actions

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Why is this needed?

This revamps our PR checks
- moves checklists out of PR description into bot-maintained comment
- fails action if bug-refs aren't confirmed or checklist items aren't completed
- moves translation template update requirement to release time

There is an extra commit here to give the action some sample data and to prevent other actions from running on this PR. ***We must remove that commit before merging.***

## Test Steps

Play with the checklist comment checkboxes and use the checkbox at the bottom of this description to re-run the actions. See how if the checklist is not properly filled out then the action fails.

---
- [x] *(un)check this to re-run the checklist action*

### CODE REVIEW

File: .github/actions/release-checklist/index.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    let ghIssues = [];
+    commits.forEach(commit => {
+        const message = commit.commit.message.toLocaleUpperCase();
+        lpBugs = lpBugs.concat(Array.from(message.matchAll(/LP: #(\d+)/g)).map(m => m[1]));
+        ghIssues = ghIssues.concat(Array.from(message.matchAll(/FIXES: #(\d+)/g)).map(m => m[1]));
+        ghIssues = ghIssues.concat(Array.from(message.matchAll(/CLOSES: #(\d+)/g)).map(m => m[1]));
+    });
+    const changelog = fs.readFileSync("./debian/changelog", { encoding: "utf8" });
+    const changelogEntries = changelog.split(/^ubuntu-advantage-tools \(/);
+    const newEntry = changelogEntries[1];
```
Comment (lucasmoura) [inline .github/actions/release-checklist/index.js:46]: Just a quick question, why are we checking the changelog here ?

**Review by lucasmoura — APPROVED** [review]
Approved, I just have one question about the changelog check, but I am assuming I am missing something here

File: .github/actions/release-checklist/index.js (hunk 1 above)
Comment (orndorffgrant) [reply .github/actions/release-checklist/index.js:46]: This is checking to see if all bugs that were referenced in commit messages are also references in the latest changelog entry (and this check is for release PRs only).

That's how [this comment](https://github.com/canonical/ubuntu-pro-client/pull/3104#issuecomment-2104766317) is generated

Comment (orndorffgrant) [reply .github/actions/release-checklist/index.js:46]: We already had this functionality in a previous release action but it was harder to use. It originated from a recurring problem of forgetting to include bug refs in the changelog in past releases.

File: debian/changelog [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ubuntu-advantage-tools (1:1+devel) UNRELEASED; urgency=medium
 
-  * wip 
+  * wip (GH: #1234, LP: #99999999)
```
Comment (renanrodrigo) [inline debian/changelog:3]: whoopsie

**Review by renanrodrigo — APPROVED** [review]

**Pull request merged**